src/ensembl/src/content/app/entity-viewer/gene-view/components/gene-function/GeneFunction.tsx

```diff
@@ -46,12 +49,38 @@ const tabClassNames = {
 
 type Props = {
   geneId: string;
+  gene: Gene;
   isNarrow: boolean;
-  selectedTabName: GeneFunctionTabName;
+  selectedTabName: GeneFunctionTabName | null;
   setActiveGeneFunctionTab: (tab: string) => void;
 };
 
 const GeneFunction = (props: Props) => {
+  const {
+    gene: { transcripts }
+  } = props;
+  let { selectedTabName } = props;
+
+  // Disable the Proteins tab if there are no transcripts data
+  // TODO: We need a better logic to disable tabs once we have the data available for other tabs
+  if (!transcripts || !transcripts.length) {
+    const proteinTabIndex = tabsData.findIndex(
+      (tab) => tab.title === GeneFunctionTabName.PROTEINS
+    );
+
+    tabsData[proteinTabIndex].isDisabled = true;
+  }
+
+  // If the selectedTab is disabled or if there is no selectedtab, pick the first available tab
```

Comment:
I wonder if we need to let the user in some way know that the tab is disabled. This is probably a UX question we'll need to ask from Anderea.

Reply:
The tab links are greyed out and will not be clickable when they are disabled.

Reply:
True, but I am wondering about the confusion that the user might be in when something else is shown instead of what he/she wanted to see.

Reply:
That's a fair point but I don't think the user will end up having a URL where the particular view is disabled.

Reply:
I think I am with Imran on this one. if there is no selected tab, I would expect us to either show a spinner (if the client is making network requests to find out which content is available) or to show a message that data is not available.

Also, I don't remember whether the Gene Function tab should be enabled if no content is available.

Reply:
>if there is no selected tab, I would expect us to either show a spinner (if the client is making network requests to find out which content is available) or to show a message that no data is available.

If no tab is selected, I think it will default to Transcripts tab. I agree that we can make a request and show a spinner when the endpoint is available for that but I am not sure what can be done now.

Also, how will the user end up having a URL for an entity that will say `?view=protein`, if there was no data for it?

> Also, I don't remember whether the Gene Function tab should be enabled if no content is available.

I think it should. Please check the following image from Andrea:
https://www.ebi.ac.uk/panda/jira/secure/attachment/34840/ev%20non-coding%20-%20gene%20function.png